core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1722,6 +1731,27 @@ object Parser extends ParserInstances {
       override def parseMut(state: State): B = Impl.map(parser, fn, state)
     }
 
+    final def select[A, B](
+        parser: Parser[Either[A, B]],
+        fn: Parser[A => B],
+        state: State
+    ): B = {
+      val cap = state.capture
+      state.capture = true
+      val either = parser.parseMut(state)
+      state.capture = cap
+      if (state.error eq null)
+        either match {
+          case Left(a) => fn.map(_(a)).parseMut(state)
```

Comment:
I think you can do fun.parseMut(state) and in the capture case actually apply the function. In the non capture case you can just return null.

I think this also solves the unmap requirement of putting in const.

Reply:
The `.asInstanceOf` in the `unmap` makes me queasy, but it makes the existing tests pass.